src/main/java/com/github/maven_nar/Compiler.java

```diff
@@ -250,6 +250,32 @@ public abstract class Compiler {
   protected Compiler() {
   }
 
+  /**
+   * Parses a "NAME=VALUE" string into a name and (optional) value. Logs to warn if argument is null, implying a blank 
+   * config item (eg &lt;define>&lt;/define>, or &lt;define>${x}&lt;/define> when x is undefined).
+   * 
+   * @param argument the string to parse (may be null)
+   * @param configName the name of the argument type, for logging purposes
+   * @return the parsed DefineArgument, or null if argument is null
+   */
+  private DefineArgument parseDefineArgument(final String argument, String configName) {
+    if (argument == null) {
+      this.mojo.getLog().warn("NAR configuration: empty " + configName + " - ignoring");
+      return null;
+    }
+    final DefineArgument define = new DefineArgument();
```

Comment:
could declare this after the following conditional that is returning null

Reply:
Good point - moved. Thanks!